src/chat.rs

```diff
@@ -215,17 +215,8 @@ impl ChatId {
         if msg.chat_id == DC_CHAT_ID_TRASH {
             return None;
         }
-        if msg.download_state != DownloadState::Done
-            // TODO (2023-09-12): Added for backward compatibility with versions that did not have
-            // `DownloadState::Undecipherable`. Remove eventually with the comment in
-            // `MimeMessage::from_bytes()`.
-            || msg
-                .error
-                .as_ref()
-                .filter(|e| e.starts_with("Decrypting failed:"))
-                .is_some()
-        {
-            // If `msg` is not fully downloaded or undecipherable, it may have been assigned to the
+        if msg.download_state == DownloadState::Undecipherable {
+            // If `msg` is not undecipherable, it may have been assigned to the
```

Comment:
Partially downloaded messages are also potentially undecipherable. The only difference is that we can download them and (likely) validate their assignment to chat. Maybe just remove this comment?